Orange/widgets/evaluate/owpredictions.py

```diff
@@ -202,24 +246,43 @@ def insert_predictor(self, index, predictor: Model):
     def remove_predictor(self, index):
         self.predictors.pop(index)
 
+    def _set_target_combos(self):
+        prob_combo = self.controls.shown_probs
+        target_combo = self.controls.target_class
+        prob_combo.clear()
+        target_combo.clear()
+
+        self._update_control_visibility()
+
+        # Set these to prevent warnings when setting self.shown_probs
+        target_combo.addItem(self.TARGET_AVERAGE)
+        prob_combo.addItems(self.PROB_OPTS)
+
+        if self.is_discrete_class:
+            target_combo.addItems(self.class_var.values)
+            prob_combo.addItems(self.class_var.values)
+            for i, tip in enumerate(self.PROB_TOOLTIPS):
+                prob_combo.setItemData(i, tip, Qt.ToolTipRole)
+            self.shown_probs = self.DATA_PROBS
+            self.target_class = self.TARGET_AVERAGE
+        else:
+            self.shown_probs = self.NO_PROBS
```

Comment:
Huh, I could but this would complicate the code and the widget.

- In terms of UX, allowing `MODEL_PROBS` when the data class is continuous but some models are categorical would require showing the combo, which would then have two options, `NO_PROBS` and `MODEL_PROBS`. When all models are regressors, the combo would be hidden...
- In terms of the code, I can now simply expect that `shown_probs` is `NO_PROBS` in case of numeric data, which simplifies some conditions, as I remember.

Though I could change this, I'd prefer not to. This situation is too strange; perhaps we could even ignore models of wrong kind and show a warning.

Reply:
Ok, leave it as is. But be aware that, in e.g. six months, one may report an issue: Predictions widget is missing probabilities. :)

Reply:
We'll tell him that he should not mix classifiers and regressors. :)